conbench/entities/run.py

```diff
@@ -84,241 +76,169 @@ def _dict_for_api_json(self) -> dict:
         return dataclasses.asdict(self)
 
 
-class Run(Base, EntityMixin):
-    __tablename__ = "run"
-    # TODO: use default uuid7 primary key
-    id: Mapped[str] = NotNull(s.String(50), primary_key=True)
-    name: Mapped[Optional[str]] = Nullable(s.String(250))
-    reason: Mapped[Optional[str]] = Nullable(s.String(250))
-    # Naive datetime object, to be interpreted in UTC. `timestamp`  is never
-    # set by API clients, i.e. the `server_default=s.sql.func.now()` is always
-    # taking effect. That also means that this property reflects the point in
-    # time of DB insertion (that should be documented in the API schema for Run
-    # objects). A more explicit way to code that would be in the create()
-    # method. The point in time by convention is stored in UTC _without_
-    # timezone information. Is a wrong point in time stored when
-    # `s.sql.func.now()` returns a non-UTC tz-aware timestamp on a DB server
-    # that does not have its system time in UTC? That should not happen, as is
-    # hopefully confirmed by the test
-    # `test_auto_generated_run_timestamp_value()`.
-    timestamp: Mapped[datetime] = NotNull(
-        s.DateTime(timezone=False), server_default=s.sql.func.now()
-    )
-    # tz-naive timestamp expected to refer to UTC time.
-    finished_timestamp: Mapped[Optional[datetime]] = Nullable(
-        s.DateTime(timezone=False)
-    )
-    info: Mapped[Optional[dict]] = Nullable(postgresql.JSONB)
-    error_info: Mapped[Optional[dict]] = Nullable(postgresql.JSONB)
-    error_type: Mapped[Optional[str]] = Nullable(s.String(250))
-
-    # The type annotation makes this a nullable many-to-one relationship.
-    # https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#nullable-many-to-one
-    # A nullable many-to-one relationship between Commit (one) and potentially
-    # many Runs, but a run can also _not_ point to a commit.
-    commit_id: Mapped[Optional[str]] = mapped_column(s.ForeignKey("commit.id"))
-    commit: Mapped[Optional[Commit]] = relationship(
-        "Commit", lazy="joined", back_populates="runs"
-    )
-
-    has_errors: Mapped[bool] = NotNull(s.Boolean, default=False)
-    hardware_id: Mapped[str] = NotNull(s.String(50), s.ForeignKey("hardware.id"))
-    hardware: Mapped[Hardware] = relationship("Hardware", lazy="joined")
-
-    # Follow https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#one-to-many.
-    # There is a one-to-many relationship between Run (one) and BenchmarkResult
-    # (0, 1, many).
-    # Ignorantly importing BenchmarkResult results in circular import err.
-    results: Mapped[List["BenchmarkResult"]] = relationship(  # type: ignore # noqa
-        back_populates="run", lazy="select"
-    )
+def _search_for_baseline_run(
+    contender_run_id: str,
+    contender_run_reason: str,
+    contender_hardware_checksum: str,
+    baseline_commit: Optional[Commit],
+    commit_limit: int = 20,
```

Comment:
Hiding whitespace differences will allow the diff of this file to be under the threshold that GitHub uses for auto-collapsing it!